server/entity/snowball.go

```diff
@@ -112,6 +113,11 @@ func (s *Snowball) New(pos, vel mgl64.Vec3, yaw, pitch float64) world.Entity {
 	return snow
 }
 
+// Explode ...
+func (s *Snowball) Explode(c block.ExplosionConfig, impact float64) {
+	s.SetVelocity(s.Position().Sub(c.Pos).Normalize().Mul(impact))
```

Comment:
This should also add on to the velocity - it shouldn't override existing velocity.